scripts/rouletteSender.js

```diff
@@ -0,0 +1,41 @@
+const PluginBtp = require('ilp-plugin-btp')
+const ILP = require('..')
+
+;(async function test () {
+  const plugin1 = new PluginBtp({ server: 'btp+ws://:plugin1@localhost:8912/' })
+  const plugin2 = new PluginBtp({ server: 'btp+ws://:plugin2@localhost:8912/' })
+  await plugin1.connect()
+  await plugin2.connect()
+  const loop = await ILP.LT.createLoop(plugin1, plugin2)
+  let cummSeen = 0
+  let numSeen = 0
+  let numAccepted = 0
+
+  async function payOnce() {
+    let destinationAmountSeen
+    const result = await loop.pay({
+      sourceAmount: '1',
+      expiresAt: new Date(new Date().getTime() + 10000),
+      loopbackHandler: (destinationAmount) => {
+        destinationAmountSeen = parseInt(destinationAmount)
+        cummSeen += destinationAmountSeen
+        numSeen++
+        // console.log({ destinationAmountSeen, threshold: (cummSeen / numSeen) })
+        return (destinationAmountSeen > (cummSeen/numSeen))
```

Comment:
I don't think you should use this equation in practice, because then if the rate goes down the payment won't be completed until the rate goes up again (which may never happen).

What code would one usually run here? And do we want to have some convenience method to send a chunked payment that does all of this automatically? I do like that you can specify the per-chunk logic manually too, though

Reply:
Yeah, and even if it stays constant, no chunks ever get accepted :) I'll improve the expression for the demo.

Rather than a convenience function, maybe we can give a few examples, or a parameterized expression where you specify a trade-off between exchange rate and speed under some assumptions about the distribution of exchange rates. If the sender has multiple ILSPs, it would also make sense to send chunks over multiple paths, to maximize exploration.